src/edu/umass/ckc/wo/ttmain/ttcontroller/TeacherToolsMainLoginController.java

```diff
@@ -31,7 +51,35 @@ public String printWelcome(@RequestParam("userName") String username, @RequestPa
             }else{
             	HttpSession session = request.getSession();
             	session.setMaxInactiveInterval(30*60);
-            	
+

```

Comment:
No direct service part need to be in the controller. Just have a look at the previous database access services, you would see-that every service is created inside -the service package and then accessed in the controller. The database access also needs to be via a separate class. Have a look at the Teacher login part and you will get to know what I an trying to Say.

Reply:
I made changes as you suggested.  It's much better now. Thanks for the advice.